flake.nix

```diff
@@ -36,23 +36,28 @@
         };
       };
 
+      cheriotPkgs = lowrisc-nix.outputs.devShells.${system}.cheriot.nativeBuildInputs;
+
       sonata-simulator = pkgs.stdenv.mkDerivation rec {
         inherit version;
         pname = "sonata-simulator";
         src = ./.;
         buildInputs = with pkgs; [libelf zlib];
         nativeBuildInputs = (with pkgs; [python311Packages.pip verilator]) ++
-          [lrPkgs.python_ot];
+          [lrPkgs.python_ot] ++ cheriotPkgs ;
         buildPhase = ''
           HOME=$TMPDIR fusesoc --cores-root=. run \
             --target=sim --tool=verilator --setup \
             --build lowrisc:sonata:system \
             --verilator_options="+define+RVFI -j $NIX_BUILD_CORES" \
-            --make_options="-j $NIX_BUILD_CORES"
+            --make_options="-j $NIX_BUILD_CORES" &&
+            make -C sw/cheri/sim_boot_stub/
```

Comment:
We probably should put the sim stub building in another derivation.

Reply:
I had a similar thought, but I'm happy for this to go in for now. In a follow up PR, we can transition to CMake and create derivations for the software builds.

Reply:
I did actually do it that way initially, then switched it over to this as it felt like it's something you should distribute with the simulator

Reply:
What could be done is to build it in a separate derivation, and then copy it to $out/share for the simulator.

Reply:
Turns out we need to do this to get the osx build to work. CHERIoT LLVM clashes with the system LLVM and the verilated C++ fails to build so will make that change.